source/fuzz/transformation_replace_id_with_synonym.cpp

```diff
@@ -57,6 +57,34 @@ bool TransformationReplaceIdWithSynonym::IsApplicable(
     return false;
   }
 
+  // If the id of interest and the synonym are integer or scalar constants with
+  // different signedness, is the instruction suitable for this replacement?
+  if (ir_context->get_def_use_mgr()->GetDef(id_of_interest)->type_id() !=
+      ir_context->get_def_use_mgr()
+          ->GetDef(message_.synonymous_id())
+          ->type_id()) {
+    // The type ids don't match
+    auto constant1 =
+        ir_context->get_constant_mgr()->FindDeclaredConstant(id_of_interest);
+    auto constant2 = ir_context->get_constant_mgr()->FindDeclaredConstant(
+        message_.synonymous_id());
+
+    if ((constant1->AsVectorConstant() &&
+         constant1->AsVectorConstant()->component_type()->AsInteger() &&
+         constant2->AsVectorConstant() &&
+         constant2->AsVectorConstant()->component_type()->AsInteger()) ||
+        (constant1->AsIntConstant() && constant1->AsIntConstant())) {
```

Comment:
... you could re-use your helper here - you could say:

if (!TypesAreEqualUpToSign(type_id_1, type_id_2) || !ITypesAreEqualUpToSign(...., ...)) {
   return false;
}

Reply:
Done